core/src/main/java/hudson/PluginManager.java

```diff
@@ -1054,6 +1054,13 @@ public HttpResponse doInstallNecessaryPlugins(StaplerRequest req) throws IOExcep
                     return RestrictiveEntityResolver.INSTANCE.resolveEntity(publicId, systemId);
                 }
 
+                @Override public void skippedEntity(String name) throws SAXException {
+                    // The underlying JAXP may have skipped resolving the entity avoiding the
+                    // exception thrown in resolveEntity. But for the sake of testing/logging
+                    // malicious behavior, go ahead and throw an exception when an entity
+                    // is skipped.
+                    throw new SAXException("Refusing to resolve entity: " + name);
```

Comment:
With `RestrictiveEntityResolver.INSTANCE.resolveEntity(...)` there is a system property to be unsafe incase there where any regressions.  It may be wise to check flag (`Boolean.getBoolean("disableXXEPrevention")`) before throwing the exception (and log otherwise).